src/metabase/db/liquibase.clj

```diff
@@ -169,9 +174,10 @@
   skip creating and releasing migration locks, which is both slightly dangerous and a waste of time when we won't be
   using them.
 
-  (I'm not 100% sure whether `Liquibase.update()` still acquires locks if the database is already up-to-date)"
-  [^Liquibase liquibase]
-  (.listUnrunChangeSets liquibase nil (LabelExpression.)))
+  IMPORTANT: this function takes `data-source` but not `liquibase` because `.listUnrunChangeSets` is buggy. See #38257."
+  [^DataSource data-source]
+  (with-liquibase [liquibase (.getConnection data-source)]
```

Comment:
I've tried to use `connection` instead of `data-source`, but it also doesn't fix the issue :)